### PR TITLE
Infiltrator suit now hides wings, antennae and obfuscates your voice

### PR DESCRIPTION
**Pull request by Xander3359**
## About The Pull Request
Infiltrator suit now hides wings, antennae and obfuscates your voice

Hidden moth wings are no longer capable of flight
## Why It's Good For The Game
Further helps the infiltrator suit hide your identity by hiding wings/antenna and makes it where you no longer have species-specific "say" verbs or lisps.
## Changelog
:cl:
qol: Infiltrator mod hides your voice
fix: Infiltrator suit now hides moth wings/antenna
fix: hidden moth wings are no longer capable of flight
/:cl:

### CODE REVIEW

File: code/modules/mod/modules/modules_antag.dm [hunk 1]
```diff
@@ -520,9 +524,16 @@
 	var/obj/item/clothing/head_cover = mod.get_part_from_slot(ITEM_SLOT_HEAD)
 	if(istype(head_cover))
 		head_cover.flash_protect = FLASH_PROTECTION_WELDER_HYPER_SENSITIVE
+	old_tongue = mod.wearer.get_organ_slot(ORGAN_SLOT_TONGUE)
+	old_tongue.forceMove(src)
```
[the remark this answers is not in the document]
Comment (Xander3359) [reply code/modules/mod/modules/modules_antag.dm:529]: got it, ty.

**Comment by GhostPlayer6942** [comment]
awesome

**Comment by carlarctg** [comment]
I'm sure there's a better way to do this than _removing the user's tongue_ temporarily.

**Comment by necromanceranne** [comment]
That's gonna get messy real quick because of that tongue swapping. I definitely think you're better off finding some way to intercept the say verbs, and just using a voice filter for tts.

File: code/modules/mod/modules/modules_antag.dm [hunk 2]
```diff
@@ -520,9 +524,17 @@
...
 		head_cover.flash_protect = FLASH_PROTECTION_WELDER_HYPER_SENSITIVE
+	old_tongue = mod.wearer.get_organ_slot(ORGAN_SLOT_TONGUE)
+	old_tongue.Remove(mod.wearer, special = TRUE)
+	old_tongue.forceMove(src)
+	robotic_voice = new()
+	robotic_voice.Insert(mod.wearer)
 
 /obj/item/mod/module/infiltrator/on_suit_deactivation(deleting = FALSE)
 	mod.wearer.remove_traits(traits_to_add, MOD_TRAIT)
+	qdel(robotic_voice)
```
Comment (Time-Green) [inline code/modules/mod/modules/modules_antag.dm:535]: Need to call Remove before you qdel here too (I believe just qdel also works but I prefer you didn’t)

**Comment by Time-Green** [comment]
> That's gonna get messy real quick because of that tongue swapping. I definitely think you're better off finding some way to intercept the say verbs, and just using a voice filter for tts.

Also agree with necromanceranne, this is not a great way to do it. You can probably intercept speech early and make sure it doesn’t get altered

File: code/modules/surgery/organs/external/wings/moth_wings.dm [hunk 3]
```diff
@@ -35,14 +35,24 @@
...
-	if(!isspaceturf(owner.loc) && !burnt)
+	if(can_fly())
 		var/datum/gas_mixture/current = owner.loc.return_air()
 		if(current && (current.return_pressure() >= ONE_ATMOSPHERE*0.85)) //as long as there's reasonable pressure and no gravity, flight is possible
 			ADD_TRAIT(owner, TRAIT_FREE_FLOAT_MOVEMENT, REF(src))
 			return
 
 	REMOVE_TRAIT(owner, TRAIT_FREE_FLOAT_MOVEMENT, REF(src))
 
+///Checks if our wings are usable
```
Comment (Mickyan) [inline code/modules/surgery/organs/external/wings/moth_wings.dm:46]: undocumented changes

Comment (necromanceranne) [reply code/modules/surgery/organs/external/wings/moth_wings.dm:46]: Its a sanity check, otherwise the wings still work even while hidden.

Comment (Mickyan) [reply code/modules/surgery/organs/external/wings/moth_wings.dm:46]: It's a change to existing items that already hide your wings, so it needs to be in the changelog

Comment (Xander3359) [reply code/modules/surgery/organs/external/wings/moth_wings.dm:46]: yeah mb

File: code/datums/components/speechmod.dm [hunk 4]
```diff
@@ -60,6 +60,8 @@
 	var/message = speech_args[SPEECH_MESSAGE]
 	if(message[1] == "*")
 		return
+	if(SEND_SIGNAL(source, COMSIG_TRY_MODIFY_SPEECH) & PREVENT_MODIFY_SPEECH)
+		return
```
Comment (Time-Green) [inline code/datums/components/speechmod.dm:64]: Could probably be a trait? I think it’s fine

**Pull request merged**